### PR TITLE
Issue #212: Add possibility to configure alpha of description text.

### PR DESCRIPTION
**Pull request by leshchenko**

### CODE REVIEW

File: taptargetview/src/main/java/com/getkeepsafe/taptargetview/TapTarget.java [hunk 1]
```diff
@@ -82,6 +82,8 @@
   boolean cancelable = true;
   boolean tintTarget = true;
   boolean transparentTarget = false;
+  // This variable is used for configuring alpha of description text.
```
Comment (xiphirx) [inline taptargetview/src/main/java/com/getkeepsafe/taptargetview/TapTarget.java]: This comment is unnecessary

File: taptargetview/src/main/java/com/getkeepsafe/taptargetview/TapTarget.java [hunk 2]
```diff
@@ -82,6 +82,8 @@
...
+  float descriptionTextAlpha = 0;
```
Comment (xiphirx) [inline taptargetview/src/main/java/com/getkeepsafe/taptargetview/TapTarget.java]: This should be initialized to 0.54f

File: taptargetview/src/main/java/com/getkeepsafe/taptargetview/TapTarget.java [hunk 3]
```diff
@@ -414,6 +416,17 @@ public TapTarget targetRadius(int targetRadius) {
     return this;
   }
 
+  /**
+  * @param descriptionTextAlpha is used to define alpha of description text, should be [0..255].
```
Comment (xiphirx) [inline taptargetview/src/main/java/com/getkeepsafe/taptargetview/TapTarget.java]: The range is [0, 1]

File: taptargetview/src/main/java/com/getkeepsafe/taptargetview/TapTarget.java [hunk 4]
```diff
@@ -414,6 +416,17 @@ public TapTarget targetRadius(int targetRadius) {
...
+  * @return TapTarget object with defined alpha for description text.
+  */
```
Comment (xiphirx) [inline taptargetview/src/main/java/com/getkeepsafe/taptargetview/TapTarget.java]: Lets just make this comment `/** Specify the alpha value [0.0, 1.0] of the description text **/`

File: taptargetview/src/main/java/com/getkeepsafe/taptargetview/TapTarget.java [hunk 5]
```diff
@@ -414,6 +416,17 @@ public TapTarget targetRadius(int targetRadius) {
...
+  public TapTarget descriptionTextAlpha(float descriptionTextAlpha) {
```
Comment (xiphirx) [inline taptargetview/src/main/java/com/getkeepsafe/taptargetview/TapTarget.java]: We should move this function to line 325

File: taptargetview/src/main/java/com/getkeepsafe/taptargetview/TapTarget.java [hunk 6]
```diff
@@ -414,6 +416,17 @@ public TapTarget targetRadius(int targetRadius) {
...
+    if (descriptionTextAlpha < 0 || descriptionTextAlpha > 255) {
```
Comment (xiphirx) [inline taptargetview/src/main/java/com/getkeepsafe/taptargetview/TapTarget.java]: < 0f and > 1f

File: taptargetview/src/main/java/com/getkeepsafe/taptargetview/TapTarget.java [hunk 7]
```diff
@@ -414,6 +416,17 @@ public TapTarget targetRadius(int targetRadius) {
...
+      throw new IllegalArgumentException("Alpha is out of allowed bounds");
```
Comment (xiphirx) [inline taptargetview/src/main/java/com/getkeepsafe/taptargetview/TapTarget.java]: `throw new IllegalArgumentException("Given an invalid alpha value: " + alpha);`

File: taptargetview/src/main/java/com/getkeepsafe/taptargetview/TapTargetView.java [hunk 8]
```diff
@@ -665,7 +665,11 @@ protected void onDraw(Canvas c) {
 
       if (descriptionLayout != null && titleLayout != null) {
         c.translate(0, titleLayout.getHeight() + TEXT_SPACING);
-        descriptionPaint.setAlpha((int) (0.54f * textAlpha));
+        if (target.descriptionTextAlpha != 0) {
+            descriptionPaint.setAlpha((int) target.descriptionTextAlpha);
+        } else {
+            descriptionPaint.setAlpha((int) (0.54f * textAlpha));
+        }
```
Comment (xiphirx) [inline taptargetview/src/main/java/com/getkeepsafe/taptargetview/TapTargetView.java]: This should now just be `descriptionPaint.setAlpha((int) (target.descriptionTextAlpha * textAlpha));`

**Comment by leshchenko** [comment]
@xiphirx Thanks for the review. I made needed changes and push new commit, could you please have a look?

**Comment by xiphirx** [comment]
Thanks!

**Comment by leshchenko** [comment]
@xiphirx Could you please update repository on jcenter() that other developers can easily use library with latest changes?

**Pull request merged**